lib/asciidoctor/chart/docinfo_processor.rb

```diff
@@ -42,9 +35,55 @@ class DocinfoProcessor < ::Asciidoctor::Extensions::DocinfoProcessor
         </style>
       HTML
 
-      def process doc
+      def process(doc)
         engines = doc.x_chart[:engines]
-        (engines.map {|engine| ASSETS[engine.to_sym] }.to_a + [DEFAULT_STYLE]).join("\n")
+        result = []
+        engines.each do |engine|
+          # noinspection RubyCaseWithoutElseBlockInspection
+          case engine
+          when 'c3js'
+            result << c3js(doc)
+          when 'chartjs'
+            result << chartjs(doc)
+          when 'chartist'
+            result << chartist(doc)
+          end
+        end
+        result << DEFAULT_STYLE
+        result.join "\n"
+      end
+
+      private
+
+      def get_path(doc, attr_name, default_path, asset_to_include)
+        doc.normalize_web_path asset_to_include, (doc.attr attr_name, default_path), false
+      end
+
+      def create_script_directive(doc, attr_name, default_path, asset_to_include)
+        %(<script src="#{get_path doc, attr_name, default_path, asset_to_include}"></script>)
+      end
+
+      def create_link_css_directive(doc, attr_name, default_path, asset_to_include)
+        %(<link rel="stylesheet" href="#{get_path doc, attr_name, default_path, asset_to_include}">)
+      end
+
+      def chartjs(doc)
+        create_script_directive(doc, CHARTJS_DIR_ATTR, DEFAULT_CHARTJS_PATH, 'chart.min.js')
+      end
```

Comment:
Not sure if this is idiomatic Ruby... 🤔 
What do you think @mojavelinux?

Also, we duplicate a lot of parameters, we could something like:

```rb
chart_assets = ChartAssets.new(doc, CHARTIST_DIR_ATTR, DEFAULT_CHARTIST_PATH)
chart_assets.create_link_css_directive('chartist.min.css')
chart_assets.create_script_directive('chartist.min.js')
```

But again not sure if this is a good practice in Ruby.
Should we mark these methods are private?